manifests/repo.pp

```diff
@@ -42,6 +48,21 @@
 
   $enabled_num = bool2num(str2bool("${enabled}")) # lint:ignore:only_variable_string
 
+  if $::osfamily == 'Debian' {
```

Comment:
What about moving this apt specific section into the apt case below in the functionality section ? That will reduce the needed code to line 52 to 56 only :)